crates/authentication/src/sys/linux/mod.rs

```diff
@@ -0,0 +1,164 @@
+use std::{collections::HashMap, sync::OnceLock};
+use zbus::{blocking::Connection, fdo, Error as ZbusError};
+use zbus_polkit::policykit1::{AuthorityProxyBlocking, AuthorizationResult, CheckAuthorizationFlags, Subject};
+
+use crate::{Error, Result, Text};
+
+pub(crate) type RawContext = ();
+
+#[derive(Debug)]
+pub(crate) struct Context;
+
+impl Context {
+    #[inline]
+    pub(crate) fn new(_: RawContext) -> Self {
+        Self
+    }
+
+    pub(crate) fn authenticate<F>(
+        &self,
+        _message: Text,
+        policy: &Policy,
+        callback: F,
+    ) -> Result<()>
+    where
+        F: Fn(Result<()>) + Send + 'static,
+    {
+        let action_id = policy.action_id;
+        // CheckAuthorization may block (D-Bus round-trips and potential user interaction).
+        // Run it off the caller thread to avoid blocking UI/event-loop threads.
+        std::thread::Builder::new()
+            .name("robius-authentication-polkit".into())
+            .spawn(move || {
+                let res = do_polkit_check(action_id);
+                callback(res);
+            })
+            .map_err(|_| Error::Unavailable)?;
+        Ok(())
+    }
+}
+
+static SYSTEM_CONNECTION: OnceLock<Result<Connection>> = OnceLock::new(); // Cached system bus connection
+static AUTHORITY_PROXY: OnceLock<Result<AuthorityProxyBlocking<'static>>> = OnceLock::new(); // Cached authority proxy
```

Comment:
Question: is it safe/correct to cache these values, or do we need to re-obtain them every time we want to show a new auth prompt?

I only ask because many different system states (like channels/sockets) change frequently, so you shouldn't necessarily cache them because the old values might no longer be valid when you want to use them next time.

For ex, if the app runs for a very long time, perhaps across sleeping/hibernating of the OS, then things like dbus connections may not be valid. Please test this out.

In general, I'd probably recommend _not_ caching these values since it doesn't offer much benefit, and may behave differently/unexpectedly across varying Linux distros.